src/admin.rs

```diff
@@ -332,15 +358,24 @@ async fn show_stats(stream: &mut OwnedWriteHalf) -> Result<(), Error> {
     let mut res = BytesMut::new();
     res.put(row_description(&columns));
 
-    let mut row = vec![
-        String::from("all"), // TODO: per-database stats,
-    ];
+    for shard in 0..pool.shards() {
+        for server in 0..pool.servers(shard) {
+            let address = pool.address(shard, server);
+            let stats = match stats.get(&address.id) {
+                Some(stats) => stats.clone(),
+                None => HashMap::new(),
```

Comment:
Is this empty hashmap only to not have to deal with no stats existing for a server?

Reply:
Yeah this is just how `get()` works to be safe: this HashMap will have the object, but Rust doesn't know that and can't validate it so we return an empty HashMap.